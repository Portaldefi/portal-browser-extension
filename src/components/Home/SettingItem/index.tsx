import React, { ReactNode } from 'react';
import { Grid, Header } from 'semantic-ui-react';

interface ISettingItem {
  name: string,
  extra: ReactNode | string | undefined;
};

export default ({name, extra}: ISettingItem) => {
  return (
    <Grid className='p-16 py-12 bg-white w-100'>
      <Grid.Column width='eleven' textAlign='left'>
        <Header as='p' className='description'>{name}</Header>
      </Grid.Column>
      <Grid.Column width='five' textAlign='right'>
        { extra }
      </Grid.Column>
    </Grid>
  )
}