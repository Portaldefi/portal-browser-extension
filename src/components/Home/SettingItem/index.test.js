import SettingItem from './index';
import { create, act } from 'react-test-renderer';

describe('SettingItem', () => {
  it('should work', () => {
    let tree;
    act(() => {
      tree = create(
        <SettingItem />
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});