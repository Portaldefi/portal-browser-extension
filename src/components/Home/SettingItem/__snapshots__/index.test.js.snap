// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingItem should work 1`] = `
<div
  className="ui grid p-16 py-12 bg-white w-100"
>
  <div
    className="left aligned eleven wide column"
  >
    <p
      className="ui header description"
    />
  </div>
  <div
    className="right aligned five wide column"
  />
</div>
`;
